jolt-core/src/r1cs/spartan.rs

```diff
@@ -836,3 +860,209 @@ impl<F: JoltField, ProofTranscript: Transcript> BatchableSumcheckInstance<F, Pro
         batched_eval_at_shift_r * eq_plus_one_shift_sumcheck
     }
 }
+
+impl<F: JoltField, ProofTranscript: Transcript> Default
+    for UniformSpartanProof<F, ProofTranscript>
+{
+    fn default() -> Self {
+        Self {
+            outer_sumcheck_proof: SumcheckInstanceProof::new(vec![]),
+            outer_sumcheck_claims: (F::zero(), F::zero(), F::zero()),
+            inner_sumcheck_proof: SumcheckInstanceProof::new(vec![]),
+            shift_sumcheck_proof: SumcheckInstanceProof::new(vec![]),
+            claimed_witness_evals: vec![],
+            shift_sumcheck_witness_eval: vec![],
+            _marker: PhantomData,
+        }
+    }
+}
+
+impl<
+        F: JoltField,
+        ProofTranscript: Transcript,
+        PCS: CommitmentScheme<ProofTranscript, Field = F>,
+    > SumcheckStages<F, ProofTranscript, PCS> for UniformSpartanProof<F, ProofTranscript>
```

Comment:
nit: would prefer to impl SumcheckStages on a separate empty struct rather than on the proof struct